lib/types.d.ts

```diff
@@ -39,8 +39,8 @@ export type KnownBrowsersKeys = keyof typeof KnownBrowsers;
 export type KnownPlatformsKeys = keyof typeof KnownPlatforms;
 
 export interface BrowserInfo {
-  name: ValueOf<KnownBrowsersKeys> | null;
-  platform: ValueOf<KnownPlatformsKeys> | null;
+  name: ValueOf<typeof KnownBrowsers> | null;
```

Comment:
thanks this was written in earlier days. The intention/reason it was written this way is to make it strict. its a string but value can be only on the known browsers. hence rather than just be any string. and can be asserted to string if needed.

I am not a typescript expert, but I feel something missing here still, or its just me. let me check once more on this.

Reply:
It _can't_ be anything other than string because you explicitly `declare const` to have strings:
https://github.com/sibiraj-s/browser-dtector/blob/204df549ed20348d8eb9602da117752461e1058d/lib/types.d.ts#L2

That said, it eludes me why is `KnownBrowsers` duplicated in `constants.ts` and `types.d.ts` (and why `types.d.ts` exists at all, for that matter). If you were reusing the actual `KnownBrowsers` (and had defined it with `as const`) the types would be more specific indeed.

Reply:
>and why types.d.ts exists at all

true, types definitions are added at a later point, and then the project is converted to typescript. it needs a cleanup in whole I guess.

Reply:
Thanks will merge this and make the cleanup mentioned and publish a release.